Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInputField.swift

```diff
@@ -0,0 +1,195 @@
+            applyTheme(theme: themeVal.theme)
+        }
+        .onChange(of: themeVal) { val in
+            applyTheme(theme: val.theme)
+        }
+    }
+
+    func applyTheme(theme: Theme) {
+        let color = theme.colors
+        errorColor = Color(color.textWarning)
+        titleColor = Color(color.textPrimary)
+        textFieldColor = Color(color.textSecondary)
+        backgroundColor = Color(color.layer2)
+    }
+
+    @ViewBuilder private func provideInputField() -> some View {
+        Text(fieldHeadline)
+            .font(.subheadline)
+            .foregroundColor(titleColor)
+        TextField("", text: $text)
+            .font(.body)
+            .padding(.top, 7.5)
+            .foregroundColor(textFieldColor)
+            .keyboardType(keyboardType)
+            .onChange(of: text) { [oldValue = text] newValue in
+                switch inputType {
+                case .name:
+                    guard !newValue.isEmpty else {
+                        inputViewModel.nameIsValid = false
+                        return
+                    }
+
+                    inputViewModel.nameOnCard = newValue
+                case .number:
+                    guard newValue.count >= 13 else {
+                        inputViewModel.numberIsValid = false
+                        return
+                    }
+
+                    guard !(text.count > userInputLimit) else {
+                        text = oldValue
+                        return
+                    }
+
+                    inputViewModel.cardNumber = newValue
+                case .expiration:
+                    guard newValue.removingOccurrences(of: " / ") != oldValue else { return }
+
+                    let numbersCount = countNumbersIn(text: text)
+
+                    guard !(text.count > formattedTextLimit) || !(numbersCount > 4) else {
+                        text = oldValue
+                        return
+                    }
+
+                    guard numbersCount % 4 == 0 else {
+                        text = text.removingOccurrences(of: " / ")
+                        return
+                    }
+
+                    inputViewModel.expirationDate = text.removingOccurrences(of: " / ")
+
+                    guard let formattedText = separate(inputType: inputType,
+                                                       for: text.removingOccurrences(of: " / ")) else { return }
+                    text = formattedText
+                }
+            }
+    }
+
+    @ViewBuilder private func errorViewWith(errorString: String) -> some View {
+        HStack(spacing: 0) {
+            Image(ImageIdentifiers.errorAutofill)
+                .renderingMode(.template)
+                .foregroundColor(errorColor)
+                .accessibilityHidden(true)
+            Text(errorString)
+                .errorTextStyle(color: errorColor)
+        }
+        .padding(.top, 7.4)
+    }
+
+    private func countNumbersIn(text: String) -> Int {
```

Comment:
Unit test for this one too 🙏, no need to set to public it could be tested trough its usage on the onChange it will be easier if the logic of the onChange was a separate function

Reply:
I think making it public would make it easier and faster to write tests for, as opposed to simulating an `onChange` on this inputField and intercepting that in tests.